log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java

```
@@ -767,6 +785,18 @@ protected void doConfigure() {
         setParents();
     }
 
+    private List<URI> convertToJavaNetUris(final List<Uri> uris) {
+        final List<URI> javaNetUris = new ArrayList<>();
+        for (Uri uri : uris) {
+            try {
+                javaNetUris.add(new URI(uri.getUri()));
+            } catch (URISyntaxException e) {
+                LOGGER.error("Error parsing monitor URI: " + uri, e);
```

Are we sure we want to proceed with (re)configuration obtained from an invalid configuration? I think we should propagate the exception and let it fail.

---

My thinking here was to be consistent with other error scenarios, for example lines 726, 764, 790.
Also to let the exception propagate we would need to declare it on the doConfigure() method which could cause an impact on anyone with their own custom implementation which extends AbstractConfiguration

---

See [this recent discussion on how to handle configuration failures](https://lists.apache.org/thread/h2oydyk6xld47ljttqvflbt4530o73vw). In short, configuration failures should result in the entire configuration to fail, instead of applying in partially. Existing code that doesn't adhere to this needs to be fixed, but that is another issue.